docs/development/validation-guidelines.md

```diff
@@ -0,0 +1,191 @@
+  - The [`cidrvalidation.CIDR`](https://pkg.go.dev/github.com/gardener/gardener/pkg/utils/validation/cidr#CIDR) interface contains several utility functions for working with CIDR ranges. An implementation of the interface can be instantiated using the [`cidrvalidation.NewCIDR`](https://pkg.go.dev/github.com/gardener/gardener/pkg/utils/validation/cidr#NewCIDR) function.
+  - The package also contains validation utilities for checking if Shoot networks intersect with Seed networks or if Shoot networks intersect between each other.
+- [`github.com/gardener/gardener/pkg/utils/validation/kubernetes/core`](https://pkg.go.dev/github.com/gardener/gardener/pkg/utils/validation/kubernetes/core), aliased as `kubernetescorevalidation`
+  - The package contains copy of several utility functions from [k8s.io/kubernetes/apis/core/validation](https://pkg.go.dev/k8s.io/kubernetes/apis/core/validation). This is done to prevent importing the `k8s.io/kubernetes` dependency in Gardener.
+- [`github.com/gardener/gardener/pkg/utils/validation`](https://pkg.go.dev/github.com/gardener/gardener/pkg/utils/validation), aliased as `validationutils`
+  - The package provides validation utilities for component configuration APIs.
+
+Frequently used validation functions:
+- [`kubernetescorevalidation.ValidateResourceQuantityValue`](https://pkg.go.dev/github.com/gardener/gardener/pkg/utils/validation/kubernetes/core#ValidateResourceQuantityValue)
+- [`kubernetescorevalidation.ValidateTolerations`](https://pkg.go.dev/github.com/gardener/gardener/pkg/utils/validation/kubernetes/core#ValidateTolerations)
+- [`validationutils.ValidateClientConnectionConfiguration`](https://pkg.go.dev/github.com/gardener/gardener/pkg/utils/validation#ValidateClientConnectionConfiguration)
+  - The `componentbaseconfigv1alpha1.ClientConnectionConfiguration` type contains details for constructing a client against a Kubernetes cluster.
+  - Use this validation function to validate fields of type `componentbaseconfigv1alpha1.ClientConnectionConfiguration` in component configuration APIs. See [example usage](https://github.com/gardener/gardener/blob/874791cb97e96d622c994c6cc8ef4091fc9e9102/pkg/controllermanager/apis/config/v1alpha1/validation/validation.go#L22).
+- [`validationutils.ValidateLeaderElectionConfiguration`](https://pkg.go.dev/github.com/gardener/gardener/pkg/utils/validation#ValidateLeaderElectionConfiguration)
+  - The `componentbaseconfigv1alpha1.LeaderElectionConfiguration` type defines the configuration of leader election clients for components that can run with leader election enabled.
+  - Use this validation function to validate fields of type `componentbaseconfigv1alpha1.LeaderElectionConfiguration` in component configuration APIs. See [example usage](https://github.com/gardener/gardener/blob/874791cb97e96d622c994c6cc8ef4091fc9e9102/pkg/controllermanager/apis/config/v1alpha1/validation/validation.go#L23).
+
+## General Guidelines
```

Comment:
I would move this section in the beginning as the document will be most probably re-used for other components, it would be nice to start with the generic guideline and then go into g/g specifics.

Reply:
I added a separate document for validation guidelines for extensions. I hope this solves your comment.